File: umbrella-insurance-core/src/main/java/com/umbrella/insurance/core/models/entities/Faq.java
package com.umbrella.insurance.core.models.entities;

import com.umbrella.insurance.core.deserializers.TimestampDeserializer;
import com.umbrella.insurance.core.serializers.TimestampSerializer;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.OnDelete;
import org.hibernate.annotations.OnDeleteAction;

import java.sql.Timestamp;

@Getter
@Setter
@Entity
@Table(name = "faqs", schema = "public", uniqueConstraints = {
        @UniqueConstraint(name = "faqs_unique", columnNames = {"faq_name"})
})
public class Faq {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "faq_id", nullable = false)
    private Long id;

    @JsonSerialize(using = TimestampSerializer.class)
    @JsonDeserialize(using = TimestampDeserializer.class)
    @NotNull
    @Column(name = "created_date_time", nullable = false)
    private Timestamp createdDateTime;

    @NotNull
    @Column(name = "question", nullable = false, length = Integer.MAX_VALUE)
    private String question;

    @NotNull
    @Column(name = "answer", nullable = false, length = Integer.MAX_VALUE)
    private String answer;

    @NotNull
    @Column(name = "faq_name", nullable = false, length = Integer.MAX_VALUE)
    private String faqName;

    @ManyToOne(fetch = FetchType.EAGER)
    @OnDelete(action = OnDeleteAction.SET_NULL)
    @JoinColumn(name = "session_id")
    private Session session;

}

File: umbrella-insurance-react-ui/src/models/faqs/v1/Faq.ts
import { Session } from "../../users/sessions/v1/Session";

export class Faq {
    id?: number;
    createdDateTime?: string; //yyyy-mm-dd hh:mm:ss
    question?: string;
    answer?: string;
    faqName?: string;
    session?: Session;
    constructor(obj?: any) {
        if (obj) {
            if (obj?.id) {
                this.id = obj.id;
            }
            if (obj?.createdDateTime) {
                this.createdDateTime = obj.createdDateTime;
            }
            if (obj?.question) {
                this.question = obj.question;
            }
            if (obj?.answer) {
                this.answer = obj.answer;
            }
            if (obj?.faqName) {
                this.faqName = obj.faqName;
            }
            if (obj?.session) {
                this.session = new Session(obj.session);
            }
        }
    }
}